src/Framework/AspNetCoreAnalyzers/test/WebApplicationBuilder/DisallowConfigureAppConfigureHostBuilderTest.cs

```diff
@@ -19,192 +23,171 @@ public async Task ConfigurationBuilderRunsWithoutDiagnostic()
-        // Act
-        var diagnostics = await Runner.GetDiagnosticsAsync(source.Source);
+builder.Host.{|#0:ConfigureHostConfiguration(builder => builder.AddJsonFile(""foo.json"", optional: true))|};
+builder.WebHost.{|#1:ConfigureAppConfiguration(builder => builder.AddJsonFile(""foo.json"", optional: true))|};
+";
+        var fixedSource = @"
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+var builder = WebApplication.CreateBuilder(args);
+builder.Configuration.AddJsonFile(""foo.json"", optional: true);
+builder.Configuration.AddJsonFile(""foo.json"", optional: true);
+";
+       
+        var expectedDiagnostic = new[] {
+            new DiagnosticResult(DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder).WithArguments("ConfigureHostConfiguration").WithLocation(0),
+            new DiagnosticResult(DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder).WithArguments("ConfigureAppConfiguration").WithLocation(1)
+        };
 
         // Assert
-        Assert.Equal(2, diagnostics.Length);
-        var diagnostic1 = diagnostics[0];
-        var diagnostic2 = diagnostics[1];
-        Assert.Same(DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder, diagnostic1.Descriptor);
-        AnalyzerAssert.DiagnosticLocation(source.MarkerLocations["MM1"], diagnostic1.Location);
-        Assert.Equal("Suggest using WebApplicationBuilder.Configuration instead of ConfigureHostConfiguration", diagnostic1.GetMessage(CultureInfo.InvariantCulture));
-        Assert.Same(DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder, diagnostic2.Descriptor);
-        AnalyzerAssert.DiagnosticLocation(source.MarkerLocations["MM2"], diagnostic2.Location);
-        Assert.Equal("Suggest using WebApplicationBuilder.Configuration instead of ConfigureAppConfiguration", diagnostic2.GetMessage(CultureInfo.InvariantCulture));
+        await VerifyCS.VerifyCodeFixAsync(source, expectedDiagnostic, fixedSource);
+    }
+
+    [Fact]
+    public async Task ConfigureAppHostBuilderProducesDiagnosticVerifyCS()
+    {
+        // Arrange
+        var source = @"
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Configuration;
+var builder = WebApplication.CreateBuilder(args);
+builder.Host.{|#0:ConfigureAppConfiguration(builder => builder.AddJsonFile(""foo.json"", optional: true))|};
+";
+        var fixedSource = @"
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Configuration;
+var builder = WebApplication.CreateBuilder(args);
+builder.Configuration.AddJsonFile(""foo.json"", optional: true);
+";
+
+        var expectedDiagnostic = new DiagnosticResult(DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder).WithArguments("ConfigureAppConfiguration").WithLocation(0);
+
+        //Act
+        await VerifyCS.VerifyCodeFixAsync(source, expectedDiagnostic, fixedSource);
```

Comment:
Let's add a test scenario where the same `ConfigureAppConfiguration` contains multiple calls inside it.

```csharp
builder.Host.ConfigureAppConfiguration(builder =>
{
    builder.AddJsonFile("foo.json", optional: true);
    builder.AddEnvironmentVariables();
});
```

should become

```
builder.Configuration.
  .AddJsonFile("foo.json", optional: true)
  .AddEnvironmentVariables();
```